spec/models/ability_spec.rb

```diff
@@ -89,11 +89,104 @@
       let(:other_self_organized_track) { create(:track, :self_organized) }
 
       it{ should be_able_to(:manage, user) }
-
       it{ should be_able_to(:manage, registration_public) }
       it{ should be_able_to(:manage, registration_not_public) }
-      it{ should_not be_able_to(:new, Registration.new(conference_id: conference_with_closed_registration.id))}
-      it{ should_not be_able_to(:create, Registration.new(conference_id: conference_with_closed_registration.id))}
+
+      # Test for user can register or not
+      context 'when user is not a speaker with event confirmed' do
+        let(:conference) { create(:conference) }
+        let(:registration_period) { create(:registration_period) }
+
+        context 'when the registration is closed' do
+          let(:conference_with_closed_registration) { create(:conference) }
```

Comment:
What's the difference between conference and conference_with_closed_registration? They are both created by the same factory, yet you expect one to have a registration which is over, and the other one to not have a registration_period at all.

Reply:
Yes, you are right about that they both created form the same factory but there is a difference.
https://github.com/openSUSE/osem/blob/master/spec/models/ability_spec.rb#L33
That's how conference and conference_with_closed_registration are different.
The only thing is that i shouldn't have create the conference_with_closed_registration here as it is already created in https://github.com/openSUSE/osem/blob/master/spec/models/ability_spec.rb#L32